src/noLifeCycleCallRule.ts

```diff
@@ -0,0 +1,59 @@
+import { LifecycleHooks } from '@angular/compiler/src/lifecycle_reflector';
+
+import * as Lint from 'tslint';
+import * as ts from 'typescript';
+import { sprintf } from 'sprintf-js';
+import { NgWalker } from './angular/ngWalker';
+
+export class Rule extends Lint.Rules.AbstractRule {
+  public static metadata: Lint.IRuleMetadata = {
+    ruleName: 'no-life-cycle-call',
+    type: 'maintainability',
+    description: 'Disallows explicit calls to lifecycle hooks.',
+    rationale: 'Explicit calls to lifecycle hooks is confusing.',
```

Comment:
We should be more specific in saying that it is Angular's responsibility to invoke these methods.

Reply:
Hey @wKoza, I agree, could you suggest a better _rationale_?

Reply:
Just 'Explicit calls to lifecycle hooks is confusing. Invoke lifecycle hooks is the responsability of Angular'

Reply:
Let's also change this:

>Explicit calls to lifecycle hooks could be confusing. Invoke lifecycle hooks is the responsability of Angular.

My only update is "is confusing" to "could be confusing".